client/components/global/BookDetail.jsx

```diff
@@ -1,117 +1,95 @@
-import React, { useState, useEffect } from 'react';
-import {
-  Modal,
-  Image,
-  Button,
-  Table,
-  Badge,
-  Spinner,
-  Row,
-  Col
-} from 'react-bootstrap';
-import axios from 'axios';
+import React, { useState, useContext, useEffect } from 'react';
+import { Modal, Image, Button, Table, Badge, Row, Col } from 'react-bootstrap';
+import { AppContext } from '../../context/context.jsx';
+import { AuthContext } from '../../context/authContext.jsx';
+import Loader from './Loader.jsx';
 
-//Props: id from Google books (number), show & owned & inLibrary (boolean), handleClose & handleAddToLibrary & handlePurchase functions
-export default function BookDetail({
-  // id,
-  inLibrary,
-  owned,
-  handlePurchase,
-  handleAddToLibrary,
-  handleClose,
-  show,
-  book
-}) {
-  // const [data, setData] = useState({});
+//Accepts only a googleId, handleClose function and show boolean.
+export default function BookDetail({ isbn = '0312537077', show, setShow }) {
+  const user = useContext(AuthContext);
+  const { getBookDetails, purchaseBook } = useContext(AppContext);
+  const [book, setBook] = useState({});
+  const [purchased, setPurchased] = useState(false);
 
-  //Temporary test constants
-  // id = '0efCDwAAQBAJ';
-  // owned = false;
-  // inLibrary = true;
-  // handlePurchase = () => {};
-  // handleAddToLibrary = () => {};
+  useEffect(async () => {
+    let data = await getBookDetails(isbn);
+    setBook(data);
+  }, [isbn]);
```

Comment:
I'm not well versed in the info this is getting but would it be a good idea to add a conditional for isbn being defined and removing its default value to prevent an unnecessary async function from running?

Reply:
good idea